File: app/src/main/java/net/twisterrob/android/view/color/swatches/pixel/drawer/ColumnByColumnBitmapDrawer.java
package net.twisterrob.android.view.color.swatches.pixel.drawer;

import net.twisterrob.android.view.color.swatches.pixel.color.PixelColor;

public class ColumnByColumnBitmapDrawer extends BitmapDrawer {
    public ColumnByColumnBitmapDrawer(int[] bitmap, int w, int h, PixelColor pixel) {
        super(bitmap, w, h, pixel);
    }

    @Override
    protected void fillPixels() {
        int w = this.w;
        int h = this.h;
        for (int x = 0; x < w; ++x) {
            for (int y = 0; y < h; ++y) {
                bitmap[y * w + x] = pixel.getPixelColorAt(x, y);
            }
            reportProgress();
        }
    }

    public static Factory factory() {
        return new Factory() {
            @Override
            public BitmapDrawer create(int[] bitmap, int w, int h, PixelColor color) {
                return new ColumnByColumnBitmapDrawer(bitmap, w, h, color);
            }
        };
    }
}

File: feature/base/src/main/java/net/twisterrob/android/view/color/swatches/pixel/drawer/ColumnByColumnBitmapDrawer.kt
package net.twisterrob.android.view.color.swatches.pixel.drawer

import net.twisterrob.android.view.color.swatches.pixel.color.PixelColor

class ColumnByColumnBitmapDrawer(
	bitmap: IntArray, w: Int, h: Int, pixel: PixelColor
) : BitmapDrawer(bitmap, w, h, pixel) {

	override fun fillPixels() {
		val w = this.w
		val h = this.h
		for (x in 0 until w) {
			for (y in 0 until h) {
				bitmap[y * w + x] = pixel.getPixelColorAt(x, y)
			}
			reportProgress()
		}
	}

	companion object {

		@JvmStatic
		@Suppress("ConvertReferenceToLambda")
		fun factory() = BitmapDrawer.Factory(::ColumnByColumnBitmapDrawer)
	}
}
